lib/components/wallet_connect/approve_token_transfer.dart

```diff
@@ -97,29 +108,35 @@ class _ApproveTokenTransferState extends State<ApproveTokenTransfer> {
                       int targetTick = latestTick + defaultTargetTickType.value;
 
                       //Send the transaction to backend
-                      bool result = false;
+                      SignedTransaction? result;
                       if (mounted) {
                         result = await sendTransactionDialog(
                             context,
                             widget.fromID!,
                             widget.toID!,
                             widget.amount,
-                            targetTick);
+                            targetTick,
+                            inputType: widget.inputType,
+                            payload: widget.payload);
 
-                        if (!result) {
+                        if (result == null) {
                           throw "Could not send transaction";
                         }
-
                         setState(() {
                           isLoading = true;
                         });
                         //If the transaction was successful
 
                         if (mounted) {
-                          Navigator.of(context).pop(ApproveTokenTransferResult(
-                              //Return the success and tick
+                          if (widget.inputType != null) {
+                            Navigator.of(context).pop(
+                                ApproveSendTransactionResult(
+                                    transactionId: result.tansactionId));
+                          } else {
+                            Navigator.of(context).pop(
+                                ApproveTokenTransferResult(tick: targetTick));
+                          }
```

Comment:
@ahmed-tarek-salem please let's add here the trxId here too

Reply:
Done with 5200b4b, should be updated in the docs of the dApp too